helix-term/src/commands.rs

```diff
@@ -3647,6 +3648,37 @@ fn yank(cx: &mut Context) {
     exit_select_mode(cx);
 }
 
+fn yank_joined_impl(editor: &mut Editor, separator: &str, register: char) -> anyhow::Result<()> {
+    let (view, doc) = current!(editor);
+    let text = doc.text().slice(..);
+
+    let values: Vec<String> = doc
```

Comment:
The existing code does this too but this code ks allocating twice here which is a bit unnecessary/potentially slow for large selections so maybe we can fix that while we are working on this anyway.

Instead you can iterate the `Selection::slices` text.slice(range) and then `push_str` to the result. This will require special handeling for the first element but you can just use that to intalize the string: `let mut res  = slices.next,().unwrap().to_string()`

Reply:
Thanks for your feedback! I think I fixed it.

Also, about introducing a new command: since the register is already configurable, I think it should be fine to leave as is. Once the special registers for the clipboard is added, it will be able to be used as is. (and the `yank_joined_to_clipboard` command can be removed) The " register seems like a sane default to me. This also reflects the discussion in #6888.

Or did you mean the typed command? I could add an argument for which register to use?

Reply:
Oh and one more thing: there's an experimental `intersperse_with` method for iterators on nightly that can be enabled with `#![feature(iter_intersperse)]`. Since it's experimental, I didn't use it, but it would simplify some of the code. Should I just add a note in the comments or something?